### PR TITLE
Component fusion

### LINKED ISSUE #53: Component fusion
Porting the fusion with interleaving
- [x] Make the algorithm from `Fusion`
- [x] make `component_fusion` on tiling
- [x] make the strategy

### PR DESCRIPTION
**Pull request by enadeau**
Closes #53

### CODE REVIEW

**Comment by enadeau** [comment]
Closes #53

File: tilings/algorithms/fusion.py [hunk 1]
```diff
@@ -165,3 +170,165 @@ def rule(self):
                         workable=[True],
                         possibly_empty=[False],
                         constructor='other')
+
+
+class ComponentFusion(Fusion):
+    """
+    Component Fusion algorithm container class.
+
+    Fuse tiling it in can be unfused by drawing a line between any component.
```
Comment (ulfarsson) [inline tilings/algorithms/fusion.py]: "if it can be"

File: tilings/algorithms/fusion.py [hunk 2]
```diff
@@ -165,3 +170,165 @@ def rule(self):
...
+
+    Check if a fusion is valid and compute the fused tiling.
+
+    If `row_idx` is provided it attempts to fuse row `row_idx` with row
+    `row_idx+1`.
+
+    If incited `col_ids` is provided it attempts to fuse column `col_idx` with
```
Comment (ulfarsson) [inline tilings/algorithms/fusion.py]: remove "incited". Also `col_ids` -> `col_idx`

File: tilings/algorithms/fusion.py [hunk 3]
```diff
@@ -165,3 +170,165 @@ def rule(self):
...
+            return self._obstruction_fuse_counter
+        obs = (ob for ob in self._tiling.obstructions if not
+               self.is_crossing_len2(ob))
+        fuse_counter = self._fuse_counter(obs)
+        self._obstruction_fuse_counter = fuse_counter
+        return self._obstruction_fuse_counter
+
+    def obstructions_to_add(self):
+        """
+        Iterator over all the obstruction obtained by fusing obstruction of the
```
Comment (ulfarsson) [inline tilings/algorithms/fusion.py]: s

Comment (ulfarsson) [reply tilings/algorithms/fusion.py]: s

File: tilings/algorithms/fusion.py [hunk 4]
```diff
@@ -165,3 +170,165 @@ def rule(self):
...
+    def _can_fuse_set_of_gridded_perms(self, fuse_counter):
+        raise NotImplementedError
+
+    def _is_valid_count(self, count, gp):
+        raise NotImplementedError
+
+    def fusable(self):
+        """
+        Return True if adjacent rows can be viewed as one row where you draw a
+        horizontal line through the permutation.
```
Comment (ulfarsson) [inline tilings/algorithms/fusion.py]: components

File: tilings/tiling.py [hunk 5]
```diff
@@ -684,6 +687,25 @@ def fusion(self, row=None, col=None):
...
+        Fuse the tilings.
+
+        If `row` is not `None` then `row` and `row+1` are fused together.
+        If `col` is not `None` then `col` and `col+1` are fused together.
+        """
+        return self._fusion(row, col, Fusion)
+
+    def component_fusion(self, row=None, col=None):
+        """
+        Fuse the tilings in such a way that it can be infused by drawing a line
```
Comment (ulfarsson) [inline tilings/tiling.py]: unfused

**Comment by enadeau** [comment]
Fixed the s

**Pull request merged**